src/main/kotlin/commands/Nick.kt

```diff
@@ -57,16 +58,47 @@ class Nick(private val chattORE: ChattORE) : BaseCommand() {
         // Note: worth having a timeout to prevent people from changing too frequently.
         if (colors.isEmpty()) throw ChattoreException("No colors provided! Please provide 1 to 3 colors!")
         val rendered = if (colors.size == 1) {
-            val code = "<${colors.first().validateColor()}>"
-            val nickname = "${code}${player.username}<reset>"
+            val color = colors.first().validateColor();
+            val nickname = "<color:$color><username></color:$color>"
             chattORE.database.setNickname(player.uniqueId, nickname)
             nickname
         } else {
             if (colors.size > 3) throw ChattoreException("Too many colors!")
-            setNicknameGradient(player.uniqueId, player.username, *colors)
+            setNicknameGradient(player.uniqueId, *colors)
         }
         val response = chattORE.config[ChattORESpec.format.chattore].render(
-            "Your nickname has been set to $rendered".miniMessageDeserialize()
+            "Your nickname has been set to $rendered".render(mapOf(
+                "username" to Component.text(player.username)
+            ))
+        )
+        player.sendMessage(response)
+    }
+
+    @Subcommand("preset")
+    @CommandCompletion("@nickPresets")
+    fun pride(player: Player, preset: String) {
+        val format = chattORE.config[ChattORESpec.nicknamePresets][preset]
+            ?: throw ChattoreException("Unknown preset! Use /nick presets to see available presets.")
+        val rendered = format.render(mapOf("username" to Component.text(player.username)));
+        chattORE.database.setNickname(player.uniqueId, format);
+        val response = chattORE.config[ChattORESpec.format.chattore].render(
+            "Your nickname has been set to <message>".render(rendered)
+        )
+        player.sendMessage(response)
+    }
+
+    @Subcommand("presets")
+    fun presets(player: Player) {
+        var renderedPresets = ArrayList<Component>()
+        for ((name, colors) in chattORE.config[ChattORESpec.nicknamePresets]) {
+            val rendered = colors.render(mapOf("username" to Component.text(name)))
+            renderedPresets.add(rendered)
+        }
+
+        val combined = renderedPresets.reduce { all, next -> "<all>, <next>".render(mapOf("all" to all, "next" to next )) }
```

Comment:
This should be broken down into multiple lines as well.

Reply:
Is there some better way to do a comma-separated join of components like this?

Reply:
This could turn into multiple lines like this:

```kotlin
reduce { all, next ->
    ...
}
````

Reply:
I think this should work but I didn't test it :D 
```
val combined = Component.join(JoinConfiguration.commas(true), renderedPresets)
```

Reply:
that works great, actually! exactly what i wanted, thanks :3